File: easy/Happy-Numbers/Happy Numbers.cs
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            // do something with line
            ShowHappyNumber(line);
        }
    }
    
    public static void ShowHappyNumber(string line){
        if(line.Trim().Length>0)
        {
            int num = System.Convert.ToInt32(line);
            char result = GetHappyNumber(30,num);
            System.Console.WriteLine(result);
        }
    }
    
    
    public static char GetHappyNumber(int deep, int num)
    {
        if (num == 1) return '1';
        else
        { 
            if (deep == 0 || num == 0) return '0';
            else
            {
                int DigSquares = 0;
                while (num!=0)
                {
                    DigSquares += (num%10)*(num%10);
                    num = num/10;
                }
                return GetHappyNumber(deep-1,DigSquares); 
            }
        }
    }
}

File: easy/Happy-Numbers/Happy Numbers.java
import java.io.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            // Process line of input Here
            ShowHappyNumber(line);
        }
    }
    
    public static void ShowHappyNumber(String line){
        if(line.trim().length()>0)
        {
            int num = Integer.valueOf(line);
            char result = GetHappyNumber(30,num);
            System.out.println(result);
        }
    }
    
    
    public static char GetHappyNumber(int deep, int num)
    {
        if (num == 1) return '1';
        else
        { 
            if (deep == 0 || num == 0) return '0';
            else
            {
                int DigSquares = 0;
                while (num!=0)
                {
                    DigSquares += (num%10)*(num%10);
                    num = num/10;
                }
                return GetHappyNumber(deep-1,DigSquares); 
            }
        }
    }
}
